flink-tests/src/test/java/org/apache/flink/test/checkpointing/utils/SavepointMigrationTestBase.java

```diff
@@ -194,49 +187,35 @@ protected final void restoreAndExecute(
 			String savepointPath,
 			Tuple2<String, Integer>... expectedAccumulators) throws Exception {
 
-		// Retrieve the job manager
-		Await.result(cluster.leaderGateway().future(), DEADLINE.timeLeft());
+		ClusterClient<?> client = miniClusterResource.getClusterClient();
+		client.setDetached(true);
```

Comment:
As a side note which is out of scope for this issue. I think we should `deprecate` `ClusterClient#setDetached`. It should not be an attribute of the client but more of how you submit a job.